### PR TITLE
Use latest docker-sync, docker-compose version

### PR DESCRIPTION
**Pull request by birdbrained**
## Purpose

Update docker-compose and docker-sync files to use the latest versions

## Changes

- Remove volumes_from and replace with volumes in dockercompose.override.yml
- Remove dest in docker-sync.yml
- Update docker-compose.yml to version 3
- Add named volume "code"
- Remove commented out code

## Side effects

May interfere with production deploy?

## Ticket

No Ticket

### CODE REVIEW

File: docker-compose.override.yml [hunk 1]
```diff
@@ -1,100 +1,33 @@
...
-#      - container:mfr-sync
-
-#  mfr_requirements:
-#    volumes_from:
-#      - container:mfr-sync
-
-#  preprints:
-#    volumes_from:
-#      - container:preprints-sync
-#########    Use this for ember-osf linked development (with docker-sync): #############
```
Comment (sloria) [inline docker-compose.override.yml]: We should still included commented blocks to enable local development for ember-osf, waterbutler, mfr, etc.

**Comment by sloria** [comment]
Ticket here: https://openscience.atlassian.net/browse/DEV-235?jql=text%20~%20%22docker-sync%22

**Comment by sloria** [comment]
Just a drive-by comment for now. I don't have time to manually test this out yet; @birdbrained Can you test out local development with any of the linked projects locally, e.g. ember-osf, waterbutler?

**Comment by sloria** [comment]
Why was waterbutler added as a submodule?

**Comment by sloria** [comment]
Any reason we shouldn't use the default strategy, `native_osx`? https://github.com/EugenMayer/docker-sync/wiki/1.2-Upgrade-Guide#040

**Comment by sloria** [comment]
Other than my 2 concerns above, this is looking good to go.

**Comment by sloria** [comment]
I assume we'll need to update `docker-compose.ubuntu.yml` if this is merged.  

@TomBaxter , Would you be willing to update that after this is merged?

**Comment by birdbrained** [comment]
Waterbutler submodule: oops.
We were using unison before. Idk if using native_osx would break linux (it seems it would detect? not sure) and also there seems to be some bugs with it still https://github.com/EugenMayer/docker-sync/wiki/8.1.-native_osx-sync-strategy-debugging-guide. From what I can tell unison has been plenty for what we've been doing with it so far. It might be something to explore separately.

**Comment by sloria** [comment]
> We were using unison before. 

@icereval would know better, but I think we were using unison because it used to be the default. In any case, I don't think switching to native_osx. This can be merged without switching strategies.

**Comment by TomBaxter** [comment]
I'll work on the ubuntu/override file. I'm wishing now I had called it linux instead of ubuntu.

**Comment by sloria** [comment]
It's your file, man. Call it whatever you want.

**Comment by TomBaxter** [comment]
@sloria @birdbrained 
https://github.com/CenterForOpenScience/osf.io/pull/7382  docker-compose.linux.yml is ready for review.

**Pull request merged**